core/storage/statistics/gae_models.py

```diff
@@ -118,9 +126,17 @@ def get_or_create_multi(cls, exploration_id, rule_data):
                 new_entity = cls(id=entity_ids[ind], answers={})
                 entities_to_put.append(new_entity)
                 entities[ind] = new_entity
+        if entities_to_put:
+            ndb.put_multi(entities_to_put)
+
+        entity_dict = {
+            exploration_id: []
+            for exploration_id, _ in exploration_state_list
+        }
 
-        ndb.put_multi(entities_to_put)
-        return entities
+        for (exploration_id, entity) in zip(exploration_ids, entities):
+            entity_dict[exploration_id].append(entity)
+        return entity_dict.values()
```

Comment:
Are you sure that this preserves the order of the exploration/state pairs that were sent in?